src/server/graph-generation/interaction/index.js

```diff
@@ -5,154 +5,173 @@ const LRUCache = require('lru-cache');
+  let reactomeIds = mediatorIds.filter( id => id.toUpperCase().includes('REACTOME'));
+
+  return {
+    data: {
+      id: summary,
+      type,
+      source: participant0,
+      target: participant1,
+      pubmedIds,
+      pcIds,
+      reactomeIds
+    },
+    classes: readableType
   };
+};
 
-  //Fetch graph from PC
-  return pc.query(params).then(res => {
-    
-    let parsedNetwork = parse(res,geneIds);
-    //avoids errors in addMetricandFilter, return empty network
-    if(_.isEmpty(parsedNetwork))
-      return {};
-
-    let filteredNetwork = addMetricAndFilter(parsedNetwork.nodes, parsedNetwork.edges);
-    return {network: filteredNetwork};
-  }).catch((e)=>{
-    logger.error(e);
-    return 'ERROR : could not retrieve graph from PC';
-  });
-}
-
-const pcCache = LRUCache({ max: PC_CACHE_MAX_SIZE, length: () => 1 });
-
-const getInteractionGraphFromPC = cache(rawGetInteractionGraphFromPC, pcCache);
-
-/**
- * Parse the PC TXT (aka extended SIF) format to JSON
- * @param {string} data graph in txt format
- * @param {String[]} Array of query Ids
- * @return {json} JSON of graph
- */
-function parse(data, queryIDs){
-  let edgeList = [];
-  let nodeList =new Map(); //maps node id to node (to avoid duplicate entries)
-
-  if(data){
-    const dataSplit=data.split('\n\n');
-    const nodeMetadata= new Map(dataSplit[1].split('\n').slice(1).map(line =>line.split('\t')).map(line => [line[0], line.slice(1) ]));
-    dataSplit[0].split('\n').slice(1).forEach(line => {
-      const splitLine=line.split('\t');
-      const edgeMetadata = interactionMetadata(splitLine[6],splitLine[4]);
-      addInteraction([splitLine[0], splitLine[2]], splitLine[1], nodeList, edgeList, nodeMetadata, edgeMetadata, queryIDs);
-    });
-
-    //return network
-    return {nodes: nodeList, edges: edgeList};
-  }
-  return {};
-}
-
-/**
- * 
- * @param {*} network JSON containing nodes and edges that represent a network
- * @returns A network JSON with 50 nodes, sorted based on degree
- * @description The network is filtered down to the 50 nodes with largest degree.
- */
-function addMetricAndFilter(nodes,edges){
-  
-  //converts the node map into an array, sorts by degree, converts back to map
-  const filteredNodes = new Map(
-    [...nodes.entries()].sort( (a, b) => {
-      return b[1].data.metric - a[1].data.metric;
-    }).slice(0,50)
-  );
-
-  //if the filtered node map has the source and target of the edge, add it to the return network
-  //if edges aren't filtered get some serious cytoscape errors later on
-  const filteredEdges = [];
-  edges.forEach( edge => {
-    const source = edge.data.source;
-    const target = edge.data.target;
-    if(filteredNodes.has(source) && filteredNodes.has(target))
-      filteredEdges.push(edge);
-  });
+let sifText2CyJson = (sifText, sourceIds) => {
+  let parsedParts = sifText.split('\n\n');
+  let interactionsData = parsedParts[0].split('\n').slice(1);
```

Comment:
Might want to make a `getData = part => part.split('\n').slice(1)` helper function